api/test/dom-helpers.js

```diff
@@ -0,0 +1,240 @@
+
+/**
+ * Waits for an element's text content to match a given string or regular expression.
+ *
+ * @export
+ * @param {Object} args - Configuration arguments.
+ * @param {Element} args.element - The root element from which to begin searching.
+ * @param {string} [args.text] - The text to search for within elements.
+ * @param {RegExp} [args.regex] - A regular expression to match against element text content.
+ * @param {boolean} [args.multipleTags=false] - Whether to look for text across multiple sibling elements.
+ * @param {number} [args.timeout=defaultTimeout] - Time in milliseconds to wait before rejecting the promise.
+ * @returns {Promise<Element|HTMLElement|null|undefined>} - A promise that resolves to the found element or null.
+ *
+ * @example
+ * waitForText({ element: document.body, text: 'Hello', timeout: 5000 })
+ *   .then(el => console.log('Element found:', el))
+ *   .catch(err => console.log('Element not found:', err));
+ */
+export function waitForText (args) {
+  return waitFor({
+    timeout: args.timeout
+  }, () => {
+    const {
+      element,
+      text,
+      regex,
+      multipleTags
+    } = args
+
+    const elems = []
+
+    let maxLoop = 10000
+    const stack = [element]
+    // Walk the DOM tree breadth first and build up a list of
+    // elements with the leafs last.
+    while (stack.length > 0 && maxLoop-- >= 0) {
+      const current = stack.pop()
+      if (current && current.children.length > 0) {
+        stack.push(...current.children)
+        elems.push(...current.children)
+      }
+    }
+
+    // Loop over children in reverse to scan the LEAF nodes first.
+    let match = null
+    for (let i = elems.length - 1; i >= 0; i--) {
+      const node = elems[i]
+      if (!node.textContent) continue
+
+      if (regex && regex.test(node.textContent)) {
+        return node
+      }
+
+      if (text && node.textContent?.includes(text)) {
+        return node
+      }
+
+      if (text && multipleTags) {
+        if (text[0] !== (node.textContent)[0]) continue
+
+        // if equal, check the sibling nodes
+        let sibling = node.nextSibling
+        let i = 1
+
+        // while there is a potential match, keep checking the siblings
+        while (i < text.length) {
+          if (sibling && (sibling.textContent === text[i])) {
+            // is equal still, check the next sibling
+            sibling = sibling.nextSibling
+            i++
+            match = node.parentElement
+          } else {
+            if (i === (text.length - 1)) return node.parentElement
+            match = null
+            break
+          }
+        }
+      }
+    }
+
+    return match
+  })
+}
+
+/**
+ * @export
+ * @param {{
+ *   event: string | Event,
+ *   element?: HTMLElement | Element | window
+ * }} args
+ *
+ * @returns {void}
+ *
+ * @throws {Error} Throws an error if the `event` is not a string that can be converted to a CustomEvent or not an instance of Event.
+ */
+export function event (args) {
```

Comment:
Rename to `fireEvent` or `dispatchEvent`?

Reply:
I like `dispatchEvent` better than fire event.